src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -856,12 +856,29 @@ private void GeneratePropertySetter<T>(QualifiedType returnType, T decl, Class @
                 WriteStartBraceIndent();
 
                 var marshal = new CSharpMarshalManagedToNativePrinter(ctx);
-                ctx.ReturnVarName = string.Format("{0}{1}{2}",
+
+                var arrayType = field.Type as ArrayType;
+                
+                if(arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    WriteLine(string.Format("fixed ({0} __arrPtr = {1}.{2})",
```

Comment:
Please extract "__arrPtr" to a local and create it using Helpers.GeneratedIdentifier("arrPtr").